mirrord/sip/src/lib.rs

```diff
@@ -1,3 +1,31 @@
+//! Concerns MacOS' SIP (System Integrity Protection) mechanism and how to sidestep it.
+//!
+//! ### Why?
+//!
+//! SIP makes it such that the `mirrord-layer` lib cannot be dynamically loaded into the user's
+//! process on MacOS machines. This is ostensibly a security measure to prevent the malicious or
+//! unintended modification of processes, but mirrord's dynamic library loading is neither. To get
+//! around this, user's process is patched by creating a new unprotected version and signing it
+//! on the fly. Then the layer can be loaded dynmically into the unprotected copy and continue as
+//! usual. If the process is a script, the shebang must be changed to point to the patched version
+//! of the interpreter.
+//!
+//! ### Usage
+//!
+//! Used by `mirrord-cli` and `mirrord-layer`. Called into before the user process is
+//! run, during setup on MacOS machines - for example, see mirrord_layer::exec_utils::patch_if_sip
+//! and mirrord::execution::MirrordExecution::start_internal.
+//!
+//! The entry point when calling this crate is [`sip_patch`](main::sip_patch).
```

Comment:
It's also used in the exec hooks, for the case a program executes a protected binary.